src/include/intarith.h

```diff
@@ -41,6 +41,19 @@
 #define ISP2(x)		(((x) & ((x) - 1)) == 0)
 
 /*
+ * Macro for checking "v" is of type "a" aligned.
+ * Note that "a" must be a power of 2, but "v" of
+ * power of 2 of "a" is not mandatory.
+ * eg, IS_P2ALIGNED(3072, 1024) == true
+ * eg, IS_P2ALIGNED(1200, 1024) == false
+ * eg, IS_P2ALIGNED(1024, 1024) == true
+ * eg, IS_P2ALIGNED(0, 1024) == true
+ * eg, IS_P2ALIGNED(0x1234, 0x100) == false
+ * eg, IS_P2ALIGNED(0x5600, 0x100) == true
+ */
+#define IS_P2ALIGNED(v, a) (((v) & ((a) - 1)) == 0)
```

Comment:
I think this macro is just checking is v is multiple of a? that can be done in single modulo operation. That looks much simpler than having separate macro for normal value and value that is of power of 2.  Not sure how much gain it will be :(.